server/src/main/java/org/spine3/server/entity/EntityRepository.java

```diff
@@ -91,9 +108,150 @@ public E load(I id) {
+     * The resulting collection is always returned with no {@code null} values.
+     *
+     * <p>The order of objects in the result is not guaranteed to be the same as the order of IDs passed as argument.
+     *
+     * <p>In case IDs contain duplicates, the result may also contain duplicates, depending on particular
+     * implementation.
+     *
+     * <p>NOTE: The storage must be assigned before calling this method.
+     *
+     * @param ids entity IDs to search for
+     * @return all the entities in this repository with the IDs matching the given {@code Iterable}
+     */
+    @CheckReturnValue
+    public ImmutableCollection<E> loadAll(Iterable<I> ids) {
+        return loadAll(ids, FieldMask.getDefaultInstance());
+    }
+
+    /**
+     * Loads all the entities in this repository by their IDs and applies the {@link FieldMask} to each of them.
+     *
+     * <p>Acts in the same way as {@link #loadAll(Iterable)}, with the {@code FieldMask} applied to the results.
+     *
+     * <p>Field mask is applied according to
+     * <a href="https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.FieldMask>FieldMask specs</a>.
+     *
+     * <p>NOTE: The storage must be assigned before calling this method.
+     *
+     * @param ids       entity IDs to search for
+     * @param fieldMask mask to apply on entities
+     * @return all the entities in this repository with the IDs contained in the given {@code ids}
+     * @see #loadAll(Iterable)
+     */
+    @CheckReturnValue
+    public ImmutableCollection<E> loadAll(Iterable<I> ids, FieldMask fieldMask) {
+        final RecordStorage<I> storage = recordStorage();
+        final Iterable<EntityStorageRecord> entityStorageRecords = storage.readBulk(ids);
+
+        final Iterator<I> idIterator = ids.iterator();
+        final Iterator<EntityStorageRecord> recordIterator = entityStorageRecords.iterator();
+        final List<E> entities = new LinkedList<>();
+
+        while (idIterator.hasNext() && recordIterator.hasNext()) {
+            final I id = idIterator.next();
+            final EntityStorageRecord record = recordIterator.next();
+            final E entity = toEntity(id, record, fieldMask);
+            entities.add(entity);
+        }
+
+        return ImmutableList.copyOf(entities);
+    }
+
+    /**
+     * Loads all the entities in this repository.
+     *
+     * <p>NOTE: The storage must be assigned before calling this method.
+     *
+     * @return all the entities in this repository
+     * @see #loadAll(Iterable)
+     */
+    @CheckReturnValue
+    public ImmutableCollection<E> loadAll() {
+        final RecordStorage<I> storage = recordStorage();
+        final Map<I, EntityStorageRecord> recordMap = storage.readAll();
+
+        final ImmutableCollection<E> entities =
+                FluentIterable.from(recordMap.entrySet())
+                              .transform(new Function<Map.Entry<I, EntityStorageRecord>, E>() {
```

Comment:
Please have a factory method for a function. It would improve readability a lot.